src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -76,25 +64,22 @@ public RepoConfiguration(String location, String branch) throws InvalidLocationE
      * @throws InvalidLocationException if {@code location} cannot be represented by a {@code URL} or {@code Path}.
      */
     public RepoConfiguration(String location, String branch, List<String> formats, List<String> ignoreGlobList,
-            boolean isStandaloneConfigIgnored, List<String> ignoreCommitList) throws InvalidLocationException {
-        this.location = location;
+                boolean isStandaloneConfigIgnored, List<String> ignoreCommitList) throws InvalidLocationException {
+        this.location = new Location(location);
         this.branch = branch;
         this.ignoreGlobList = ignoreGlobList;
         this.isStandaloneConfigIgnored = isStandaloneConfigIgnored;
-        this.formats = formats;
-
-        validateIgnoreCommits(ignoreCommitList);
-        this.ignoreCommitList = ignoreCommitList;
+        this.formats = new FormatList(formats);
+        this.ignoreCommitList = new IgnoreCommitsList(ignoreCommitList);
 
-        verifyLocation(location);
         Matcher matcher = GIT_REPOSITORY_LOCATION_PATTERN.matcher(location);
 
         if (matcher.matches()) {
             organization = matcher.group("org");
             repoName = matcher.group("repoName");
             displayName = organization + "_" + repoName + "_" + branch;
         } else {
-            repoName = Paths.get(location).getFileName().toString().replace(GIT_LINK_SUFFIX, "");
+            repoName = Paths.get(location).getFileName().toString().replace(Location.GIT_LINK_SUFFIX, "");
```

Comment:
I think the `Paths.get(location).getFileName().toString().replace(Location.GIT_LINK_SUFFIX, "");` can be replaced with a method in Location? That way we don't need to reference the `Location.GIT_LINK_SUFFIX`.